src/pyTypeFactory.cc

```diff
@@ -107,7 +109,7 @@ PyType *pyTypeFactory(JSContext *cx, JS::Rooted<JSObject *> *global, JS::Rooted<
         break;
       }
     case js::ESClass::Function: {
-        PyObject *JSCxGlobalFuncTuple = Py_BuildValue("(lll)", (long)cx, (long)global, (long)rval);
+        PyObject *JSCxGlobalFuncTuple = Py_BuildValue("(lll)", (uint64_t)cx, (uint64_t)global, (uint64_t)rval);
         PyObject *pyFunc = PyCFunction_New(&callJSFuncDef, JSCxGlobalFuncTuple);
         returnValue = new FuncType(pyFunc);
         memoizePyTypeAndGCThing(returnValue, *rval); // TODO (Caleb Aikens) consider putting this in the FuncType constructor
```

Comment:
Consider renaming `global` to `thisv`. Only in special cases `this == globalThis`

This is not the global variable
https://github.com/Distributive-Network/PythonMonkey/blob/3c6ff101bac4a61f8e74fed0e4ccbfd178d2505c/src/jsTypeFactory.cc#L177

Reply:
Also, would `thisv` be GCed as well?

Reply:
The `obj` argument to [`JS_CallFunctionValue`](https://hg.mozilla.org/releases/mozilla-esr102/file/tip/js/public/CallAndConstruct.h#l64) is indeed the `this` value.